fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -1011,7 +1009,7 @@ public void updateLakeTableColocationInfo(OlapTable olapTable) throws DdlExcepti
     }
 
     private void constructLakeGroups(GlobalStateMgr globalStateMgr) {
-        if (!Config.use_staros) {
+        if (GlobalStateMgr.getCurrentState().isSharedNothingMode()) {
```

Comment:
use `!GlobalStateMgr.getCurrentState().isSharedDataMode()`

currently `GlobalStateMgr.getCurrentState().isSharedNothingMode()` is equivalent to `!GlobalStateMgr.getCurrentState().isSharedDataMode()`, when a third mode is introduced, the assumption doesn't hold any more.